### ISSUE TITLE
[telemetry] Count CSRF nonce checks in WebPageCSRFHandler

### ISSUE THREAD

**Issue by phax** [phax/ph-oton#47]
## Goal

Count CSRF nonce mismatches, so that a broken or attacked form is visible in monitoring rather than only in the log.

## Where

Two related sites — instrument the uicore one, and use the app one only as the low-level check:

**1. `ph-oton-uicore/src/main/java/com/helger/photon/uicore/page/WebPageCSRFHandler.java`** — the real hook:

```java
public EContinue checkCSRFNonce (@NonNull final IWebPageExecutionContext aWPEC)
```

This is where a mismatch is turned into a rejected request. It delegates to an `ICSRFErrorHandler` (default `LoggingCSRFErrorHandler`), which today only logs. `WebPageCSRFHandler.INSTANCE` is the shared default instance, and `DEFAULT_CSRF_PREVENTION_ENABLED` is `true`, so this path is live in every ph-oton application unless explicitly disabled.

**2. `ph-oton-app/src/main/java/com/helger/photon/app/csrf/CSRFSessionManager.java`** — the primitive:

```java
public boolean isExpectedNonce (@Nullable final String sNonceToCheck)
```

Prefer instrumenting (1): it has the page context, and (2) is also called in situations that are not failures.

## What to emit

Extend `com.helger.photon.uicore.CUICoreTelemetry` (create it here if the `AbstractWebPage` issue has not landed yet):

| Constant | Value |
|---|---|
| `METRIC_CSRF_CHECKS` | `photon.csrf.checks` |
| `ATTR_CSRF_VALID` | `photon.csrf.valid` |
| `ATTR_CSRF_PAGE_ID` | `photon.csrf.page.id` |
| `UNIT_CHECK` | `{check}` |

**Counter** `photon.csrf.checks`, attributes:
- `valid` — `true` / `false`
- `page.id` — from `aWPEC.getMenuItem ().getID ()` or the equivalent page ID reachable from `IWebPageExecutionContext`; verify the exact accessor before use. This is the same bounded, registered ID used by the `AbstractWebPage` instrumentation.

Counting *all* checks rather than only the failures is deliberate: a failure count without a total tells you nothing about whether 3 failures out of 4 or out of 40 000 requests.

## Cardinality

Page ID is bounded (registered menu items). Do **not** add the nonce value itself, the session ID, or the user — the nonce is a security token and must never leave the process in telemetry.

## POM

`ph-oton-uicore` gets `ph-telemetry` transitively via `ph-oton-core` → `ph-oton-mgrs`, but declare it directly since it is used directly (no `<version>` — the BOM is imported in the root POM):

```xml
<dependency>
  <groupId>com.helger.telemetry</groupId>
  <artifactId>ph-telemetry</artifactId>
</dependency>
```

## Acceptance criteria

- [ ] Both the valid and the invalid path increment the counter.
- [ ] The nonce value never appears in any attribute.
- [ ] `LoggingCSRFErrorHandler` keeps logging as before — telemetry is added alongside, not instead.
- [ ] `mvn install` passes; `README.md` news bullet added.

---

### Shared conventions for all ph-telemetry work in ph-oton

**Facades.** `com.helger.telemetry.Telemetry` for spans, `com.helger.telemetry.TelemetryMetrics` for instruments. If no `ITelemetryTracerSPI` / `ITelemetryMeterSPI` is registered by the deploying application, everything degrades to cheap no-ops — so emit unconditionally, never guard with an "is telemetry enabled" flag.

**No OpenTelemetry in ph-oton.** Only `com.helger.telemetry:ph-telemetry` is ever added. `ph-telemetry-otel` and the OpenTelemetry SDK are the deploying application's business. The version comes from the `ph-telemetry-parent-pom` BOM already imported in the root `pom.xml`, so the new `<dependency>` needs no `<version>`.

**Reference implementation.** `ph-oton-mgrs/src/main/java/com/helger/photon/mgrs/longrun/` — `CLongRunningJobTelemetry` (names), `LongRunningJobMetrics` (instrument holder), `LongRunningJobTelemetry` (package-private emitter). Follow that layout.

**Constants: one class per module** (decided; `CLongRunningJobTelemetry` stays where it is for backwards compatibility since it is public API since 10.4.0):

| Module | Constants class |
|---|---|
| ph-oton-io | `com.helger.photon.io.CIOTelemetry` |
| ph-oton-app | `com.helger.photon.app.CAppTelemetry` |
| ph-oton-audit | `com.helger.photon.audit.CAuditTelemetry` |
| ph-oton-ajax | `com.helger.photon.ajax.CAjaxTelemetry` |
| ph-oton-api | `com.helger.photon.api.CAPITelemetry` |
| ph-oton-security | `com.helger.photon.security.CSecurityTelemetry` |
| ph-oton-mgrs | `CLongRunningJobTelemetry` (existing) + `com.helger.photon.mgrs.CMgrsTelemetry` (new) |
| ph-oton-core | `com.helger.photon.core.CCoreTelemetry` |
| ph-oton-connect | `com.helger.photon.connect.CConnectTelemetry` |
| ph-oton-uicore | `com.helger.photon.uicore.CUICoreTelemetry` |
| ph-oton-datatables | `com.helger.photon.uictrls.datatables.CDataTablesTelemetry` |

Each is `@Immutable`, `final`, with a private constructor, and groups constants under `// === span names ===`, `// === metric instrument names ===`, `// === attribute keys ===`, `// === metric units ===` comments, exactly as `CLongRunningJobTelemetry` does. Every constant gets a Javadoc comment.

**Naming.** Spans and metrics `photon.<area>.<thing>`, attributes `photon.<area>.<key>`. Units follow the OpenTelemetry convention: `ms` for durations, `{request}` / `{login}` / `{job}` / `{session}` for counts.

**Cardinality is a hard rule.** Metric attributes must be bounded and small. Anything unbounded — user IDs, session IDs, concrete URL paths, execution IDs, IP addresses, exception messages — goes on **spans only**, never on a metric. This mirrors `ATTR_JOB_TYPE` (metric) vs `ATTR_JOB_EXECUTION_ID` (span-only) in `CLongRunningJobTelemetry`.

**Span/thread binding.** A span is bound to the thread that created it (see the warning block at the end of `LongRunningJobTelemetry.onJobEnd`). Prefer `Telemetry.withSpan*` so start and end are guaranteed to happen on the same thread; only fall back to a manual `startSpan` + try/finally when the span genuinely has to outlive the method.

**Existing `StatisticsManager` handlers stay.** Where a site already emits ph-commons statistics, telemetry is added *alongside* them — do not remove or change the statistics handlers. The long-term strategy for the two systems is tracked separately.

**Coding style.** phax styleguide: Hungarian notation, `final` parameters, JSpecify `@NonNull`/`@Nullable`, `_` prefix for private methods, private methods declared before their callers, space before `(` and `<`, inline string concatenation in log calls (no SLF4J `{}` placeholders), Apache 2.0 header.

**Definition of done.** `mvn install` passes; a bullet is added to the `v10.5.0 - work in progress` section of `README.md` under "News and noteworthy".

**Comment by phax** [comment]
Part of #55 — see that issue for the shared conventions, the module-level constants class table and the POM impact overview.

**Comment by phax** [comment]
Implemented for **10.5.1**.

### What was added

| File | Kind |
|---|---|
| `ph-oton-uicore/.../uicore/CUICoreTelemetry.java` | extended - `METRIC_CSRF_CHECKS`, `ATTR_CSRF_VALID`, `ATTR_CSRF_PAGE_ID`, `UNIT_CHECK` |
| `ph-oton-uicore/.../uicore/page/CSRFMetrics.java` | new, public - 1 counter |
| `ph-oton-uicore/.../uicore/page/CSRFTelemetry.java` | new, package-private emitter |
| `ph-oton-uicore/.../uicore/page/WebPageCSRFHandler.java` | modified |
| `ph-oton-uicore/src/test/.../CSRFTelemetryTest.java` | new, 2 tests |

`ph-oton-uicore` already declares `com.helger.telemetry:ph-telemetry` directly since #42, so no POM change was needed.

### Emission

Site (1) from the issue - `WebPageCSRFHandler.checkCSRFNonce (...)` - was instrumented; `CSRFSessionManager.isExpectedNonce (...)` in ph-oton-app was left alone, as advised.

The check result is now bound to a local before it is used, so both branches share one emission point:

```java
final boolean bValid = aCSRFSessionMgr.isExpectedNonce (sRetrievedNonce);
CSRFTelemetry.onCSRFNonceChecked (aWPEC, bValid);
if (!bValid)
{ ... }
```

Counter `photon.csrf.checks` with `photon.csrf.valid` (`true`/`false`) and `photon.csrf.page.id`. Valid checks are counted too, so a failure *rate* can be computed - a failure count alone does not say whether it was 3 out of 4 or 3 out of 40000 requests.

The increment sits **inside** the `isCSRFPreventionEnabled ()` guard: when CSRF prevention is switched off no check is performed, so counting one would be wrong.

### Page ID accessor

`aWPEC.getWebPage ().getID ()`. `IWebPage` extends `IPage`, so this is the very same registered, bounded page ID that `AbstractWebPage` uses for `photon.webpage.id` in #42 - the two instruments join on the same dimension. This is preferable to going through the selected menu item, which can differ from the page actually being rendered.

### The nonce never leaves the process

Only the page ID and the boolean are attributes. `CSRFTelemetryTest.testInvalidNonce` feeds a known wrong nonce and asserts both that the attribute map does not contain that value and that it has exactly 2 entries, so the test fails if anyone adds the nonce later.

`LoggingCSRFErrorHandler` is untouched - `m_aErrorHdl.onCSRFError (...)` still runs exactly as before on the failure path.

### Verification

`CSRFTelemetryTest`: 2/2 green (a valid and an invalid nonce, driven through a real `WebPageExecutionContext` and a real session-scoped `CSRFSessionManager`). A test was added even though the acceptance criteria did not require one. Full reactor `mvn install` over all 23 modules: BUILD SUCCESS. `README.md` news bullet added under `v10.5.1 - work in progress`.
